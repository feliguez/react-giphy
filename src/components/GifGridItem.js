import PropTypes from 'prop-types';

const GifGridItem = ({ title, url }) => {
  return (
    <article className="card animate__animated animate__fadeIn">
      <img alt={title} src={url} />
      <p>{title}</p>
    </article>
  );
};

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}

export default GifGridItem;
