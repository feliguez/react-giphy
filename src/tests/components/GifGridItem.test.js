import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('Test on GifGridItem', () => {
  const title = 'TÃ­tulo';
  const url = 'http://localhost/algo.jpg';
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('Component should render as its snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Image should be equal from props', () => {
    const img = wrapper.find('img');
    // console.log(img);

    expect(img.prop('alt')).toBe(title);
    expect(img.prop('src')).toBe(url);
  });

  test('Article should have .card as className', () => {
    const card = wrapper.find('article');
    const className = card.prop('className');
    // console.log(card.prop('className'));

    expect(className.includes('card')).toBe(true);
  });
});
