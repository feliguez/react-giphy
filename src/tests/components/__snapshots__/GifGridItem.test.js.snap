// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on GifGridItem Component should render as its snapshot 1`] = `
<article
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Título"
    src="http://localhost/algo.jpg"
  />
  <p>
    Título
  </p>
</article>
`;
